### ISSUE TITLE
Add all missing oauth client properties to the CRD spec

### ISSUE THREAD

**Issue by p-bakker** [ory/hydra-maester#36]
As discussed in https://github.com/ory/hydra-maester/issues/26 there are a whole bunch of oauth client properties missing from the CRD spec. 

See https://www.ory.sh/docs/hydra/sdk/api#schemaoauth2client for everything that can be set on an oauth client, at least in Hydra.

In order to make maester a production-ready solution for syncing oauth client registrations between Hydra instances, it should support all properties that an oauth client can have in Hydra

**Comment by amihalj** [comment]
this is quite a chunk of work, I believe any PR would be welcomed here.
since I am using **metadata** from the latest hydra release, I could make a PR for that some time soon.

I suggest that hydra community should give a hand here. if there is a list of priorities or critical client properties it would be better that @aeneasr places them here and anyone ready to contribute can jump in

**Comment by amihalj** [comment]
I have created https://github.com/ory/hydra-maester/pull/38 for Metadata support. step by step

**Comment by aeneasr** [comment]
Absolutely, unfortunately I have zero experience with CRD Controllers in Kubernetes. The maester efforts are purely community driven at the moment, but as you've might seen we're hiring talented engineers to help out with these projects! For myself to get involved I would probably have to clone myself :D

I will however discuss this with the people who contributed the most to these controllers, maybe we can find an interim solution!

**Comment by amihalj** [comment]
@aeneasr I have tried to make a PR https://github.com/ory/hydra-maester/pull/38 for metadata, but CI build is failing. I have tried to dig out the reason but I got stuck. If you know what could be the reason or know who could know I would appreciate it a lot.

thanks

**Comment by aeneasr** [comment]
Of course, it was on my task list already and I just replied :)

**Comment by p-bakker** [comment]
Hi,

I/we will contribute, but it'll be a few weeks before we get to it, probably early in the new year

Paul
